code/modules/reagents/Chemistry-Reagents/Drug-Reagents.dm

```diff
@@ -168,40 +168,34 @@
 	overdose_threshold = 15
 	addiction_threshold = 10
 	metabolization_rate = 0.75 * REAGENTS_METABOLISM
+	stun_threshold = 6
+	speedboost = VERY_FAST
 
 /datum/reagent/drug/methamphetamine/on_mob_life(mob/living/M)
 	var/high_message = pick("You feel hyper.", "You feel like you need to go faster.", "You feel like you can run the world.")
-	var/tolerance_message = pick("You don't feel so good.", "You feel a bit sick.", "You feel like you've had better days.")
-	if(current_cycle <= 50)
-		if(prob(5))
-			M << "<span class='notice'>[high_message]</span>"
-		M.AdjustParalysis(-2)
-		M.AdjustStunned(-2)
-		M.AdjustWeakened(-2)
-		M.adjustStaminaLoss(-2)
-		M.status_flags |= GOTTAGOREALLYFAST
-	if(current_cycle > 50)
-		if(prob(15))
-			M << "<span class='notice'>[tolerance_message]</span>"
-		M.adjustStaminaLoss(3)
-		M.adjustToxLoss(1*REM)
+
+	if(prob(5))
+		M << "<span class='notice'>[high_message]</span>"
+	if(!(M.stunned || M.weakened || M.paralysis))
+		stun_timer++
+		metabolization_rate = 0.75* REAGENTS_METABOLISM
```

Comment:
Could you not use initial(metabolization_rate) is your multiplier?
So if someone wants to change the metabolizationrate, they only need to change the value at the definitions of the reagent?